geode/src/test/java/org/apache/calcite/adapter/geode/rel/JsonLoader.java

```diff
@@ -51,12 +53,20 @@
   private void load(Reader reader) throws IOException {
     Objects.requireNonNull(reader, "reader");
     try (BufferedReader br = new BufferedReader(reader)) {
-      int key = 0;
+      List<Map> mapList = new ArrayList<>();
       for (String line; (line = br.readLine()) != null;) {
         Map jsonMap = mapper.readValue(line, Map.class);
-        PdxInstance pdxInstance = mapToPdx(rootPackage, jsonMap);
-        region.put(key++, pdxInstance);
+        mapList.add(jsonMap);
       }
+      loadMapList(mapList);
+    }
+  }
+
+  void loadMapList(List<Map> mapList) {
```

Comment:
Now when I think, maybe it's better to have [ObjectNode](https://fasterxml.github.io/jackson-databind/javadoc/2.8/com/fasterxml/jackson/databind/node/ObjectNode.html) instead of Map. It it closer to JSON format.

You can make another method:
```java
loadAll(Iterable<ObjectNode> data); // OR 
loadAll(Stream<ObjectNode> data);
```

Reply:
ObjectNode is not supporting to put java.sql objects like Date, Time, Timestamp which I am testing in the new Test class. I am able to create those objects with Map.